### ISSUE TITLE
[BUG] Ignoring nested schema examples

### ISSUE THREAD

**Issue by ravy** [schemathesis/schemathesis#2358]
### Checklist

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

When trying out examples in swagger 2.0 i see that the example values of nested objects are not considered

### To Reproduce

🚨 **Mandatory** 🚨: Steps to reproduce the behavior:

1. Run this command `st run -v --hypothesis-phases=explicit --validate-schema=true -c all -H "Authorization: Bearer root" --generation-allow-x00=False --hypothesis-verbosity=debug --base-url=http://localhost:8083 tree-openapi202.json --show-trace --dry-run`
2. See error

Please include a minimal API schema causing this issue:
minimal api schema attached (see Additional context section)

### Expected behavior
lets say we have a hypothetical tree object with related birds that live on the tree
```
tree:
    name: birch
    height: 25m
    bird: 
       name: bird1
       description: example-bird
```
Assume name fields are mandatory in tree, bird
Expect to see these examples.
1. `{ tree { name: birch }}` Minimal example with only required fields of only tree
2. `{ tree: { name: birch, bird: { name: bird1 }}` Minimum example with only required fields of tree, bird
3. `{ tree: { name: birch, height: 25m, bird: { name: bird1, description: example-bird }}` Maximum example with all fields populated

examples 2, 3 are missing in schemathesis
schemathesis never generates any example of bird under the tree whether or not we declare name fields as mandatory
note: dredd behavior is only example 3, greedy approach

### Environment

`platform Linux -- Python 3.9.18, schemathesis-3.33.0, hypothesis-6.103.5, hypothesis_jsonschema-0.23.1, jsonschema-4.22.0
`

### Additional context
this is the sample swagger doc used for testing
[tree-openapi20.json](https://github.com/user-attachments/files/16315777/tree-openapi20.json)

**Comment by ravy** [comment]
this seems more severe, not just swagger 2.0, this does not work for openapi 3.0 too, changed title

**Comment by Stranger6667** [comment]
Thank you for opening this issue!

The problem is that Schemathesis does not look behind references when filling in fields for examples. It should be more or less straightforward to fix except for recursive cases.
The overall approach is to yield the minimum number of complete examples that cover all explicit individual examples specified in the schema. The required / optional distinction is somewhat arbitrary from the JSON Schema point of view (as it is just another validator), hence I'd prefer to keep the current direction and do not explicitly generate all combinations of required / optional. I.e. have the greedy approach as dredd

**Comment by Stranger6667** [comment]
Fixed in `3.33.1`
